src/cli/cli.cpp

```diff
@@ -1847,6 +1853,109 @@ void Interpreter::ProcessLeaderWeight(uint8_t aArgsLength, char *aArgs[])
 }
 #endif // OPENTHREAD_FTD
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+void Interpreter::HandleLinkMetricsReport(const otIp6Address *       aAddress,
+                                          const otLinkMetricsValues *aMetricsValues,
+                                          void *                     aContext)
+{
+    static_cast<Interpreter *>(aContext)->HandleLinkMetricsReport(aAddress, aMetricsValues);
+}
+
+void Interpreter::HandleLinkMetricsReport(const otIp6Address *aAddress, const otLinkMetricsValues *aMetricsValues)
+{
+    OutputFormat("Received Link Metrics Report from: ");
+    OutputIp6Address(*aAddress);
+    OutputFormat("\r\n");
+
+    if (aMetricsValues->mMetrics.mPduCount)
+    {
+        OutputFormat(" - PDU Counter: %d %s\r\n", aMetricsValues->mPduCountValue,
+                     otLinkMetricsTypeEnumToString(OT_LINK_METRICS_TYPE_COUNT));
```

Comment:
These strings are also used in DebgLog in `link_metrics.cpp`. What about defining them in `link_metrics.cpp` so that both `cli` and core and use them?

Reply:
`cli` is application layer and technically should only see and use public OT definitions and API (I know there are certain cases where this is not followed but I think we should address and fix them over time).

I won't recommend breaking the layers, to allow sharing strings between core and CLI (for small code-size optimization).

In my opinion it's fine to have the strings defined spcpeifc to each layer, i.e. CLI has its own definition and we add it it in core in debug log only - note that the log debug string are optimized out when debug level is not used - commonly used log levels are info, note, warn and debg level is too verbose and intended for debugging)